cobra/ui/interface.go

```diff
@@ -0,0 +1,57 @@
+/*
+ *  MIT License
+ *
+ *  Copyright (c) 2020 Nicolas JUHEL
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ */
+
+package ui
+
+import "github.com/spf13/cobra"
+
+type Question struct {
+	Text         string
+	Options      []string
+	Handler      func(string) error
+	PasswordType bool
+	FilePath     bool
```

Comment:
add option to custom contents following input
like spefic function by key pressed : 
FuncKeyDown
FuncKeyUp
FuncUpdate...

as that if function are not set, use standard run, but if user defined set, call the user defined function

Reply:
For instance just the cursor and the question color are allowed to be specified by the user for each question, I think we most keep the the UP - DOWN - BACKSPACE as they are as well or if you agree we can at least let the user to choose them explicitly but the code logic (the behind functionalities will remain intact)